### PR TITLE
output transform

### PR DESCRIPTION
**Pull request by goodwanghan**
`OutputTransformer`, `OutputCoTransformer`, `Transformer`, `CoTransformer`, or transformer like functions with or without decorator, they are all lazy in Fugue (meaning that Fugue doesn't guarantee an immediate execution of the transformers)

Now, we have a new feature in FugueWorkflow and SQL, `out_transform`, it has two major differences compared to `transform`
* It is not lazy, Fugue guarantees it runs immediately when reaching the step
* It has no output. It is the final step of a branch in the execution graph.

Alternatively, you may use `transform().persist()` to achieve the similar goal, but
* `persist` or `checkpoint` will take either memory or disk space, `out_transform` takes no memory or space
* `out_transform` can be used with anything `transform` can use, plus, it can take functions with `None` output as output transformers. So it is more convenient.

### CODE REVIEW

File: fugue_test/builtin_suite.py [hunk 1]
```diff
@@ -435,6 +461,157 @@ def test_cotransform_with_key(self):
                 e = dag.df([[1, 3, 1]], "a:int,ct1:int,p:int")
                 e.assert_eq(c)
 
+        def test_out_transform(self):  # noqa: C901
```
Comment (goodwanghan) [inline fugue_test/builtin_suite.py:464]: This is the test showing the purpose of out_transform

File: fugue_test/builtin_suite.py [hunk 2]
```diff
@@ -435,6 +461,157 @@ def test_cotransform_with_key(self):
...
+                a.out_transform(t8, ignore_errors=[NotImplementedError])
+                raises(FugueWorkflowCompileValidationError, lambda: a.out_transform(t2))
+                raises(FugueWorkflowCompileValidationError, lambda: a.out_transform(t3))
+                raises(FugueWorkflowCompileValidationError, lambda: a.out_transform(t4))
+                raises(FugueWorkflowCompileValidationError, lambda: a.out_transform(t5))
+                raises(FugueWorkflowCompileValidationError, lambda: a.out_transform(T7))
+
+            assert 11 <= incr()
+
+        def test_output_cotransform(self):  # noqa: C901
```
Comment (goodwanghan) [inline fugue_test/builtin_suite.py:541]: This is the test showing the purpose of out_transform

File: tests/fugue_sql/test_workflow_parse.py [hunk 3]
```diff
@@ -180,6 +184,38 @@ def test_transform():
     )
 
 
+def test_out_transform():
```
Comment (goodwanghan) [inline tests/fugue_sql/test_workflow_parse.py:187]: This is the SQL test showing the purpose of out_transform

**Comment by goodwanghan** [comment]
test coverage is still 100%

**Review by kvnkho — APPROVED** [review]

**Pull request merged**